nautobot/core/ui/object_detail.py

```diff
@@ -894,6 +894,79 @@ def render_body_content(self, context):
         return result
 
 
+class StatsPanel(Panel):
+    def __init__(
+        self,
+        *,
+        filter_name,
+        filter_pks=None,
+        related_field_names=None,
+        body_content_template_path="components/panel/stats_panel_body.html",
+        **kwargs,
+    ):
+        """Instantiate a `StatsPanel`.
+
+        Args:
+            related_field_names (dict): key/value pair of { <model_class>: <query_strings> }.
+            filter_pks (list of uuids): the list of pks work in tandem with the query_strings.
+            filter_name (str): a valid query filter append to the anchor tag for each stat button.
+        """
+
+        self.filter_name = filter_name
+        self.related_field_names = related_field_names
+        self.filter_pks = filter_pks
```

Comment:
Is this always overwritten by lines 937-942 below? Not sure I immediately see the application of providing this at declaration time.